### ISSUE TITLE
Cross-compiling bindbc-sdl (1.4.7) for Android (aarch64-*-linux-android)

### ISSUE THREAD

**Issue by DannyArends** [BindBC/bindbc-common#2]
I was upgrading my bindbc dependencies (SDL, openGL, etc) to the latest version for my [CalderaD](https://github.com/DannyArends/CalderaD) project, and ran into an issue using the newest versions of the bind-bc common library.

When cross compiling using LDC2 (targeting Android, aarch64-*-linux-android) I now get an **unknown runtime** error:

```
../../.dub/packages/bindbc-common/0.1.5/bindbc-common/source/bindbc/common/codegen.d(420,8): Error: static assert:  "Unknown runtime, not sure what mangling to use. Please check how your compiler mangles C++ struct constructors and add a case for it to `bindbc.common.codegen.mangleofCppDefaultCtor`."
```

Previous versions of the bindbc-loader, e.g. v0.3.2, did not have a dependency on bindbc-core, but cross-compiled to Android aarch64 perfectly fine without any issues.

 How would I go about to solve this ?

**Comment by DannyArends** [comment]
For reference, this is the dub.json that I use to compile an empty main file:

```json
{
	"name": "bindbctest",
	"description": "BindBC test",
	"authors": ["Danny Arends"],
	"copyright": "Danny Arends",
	"license": "Unknown",
	"homepage": "http://www.dannyarends.nl",
	"sourcePaths": ["src"],
	"configurations": [
	{
		"name": "android-64",
		"targetType": "dynamicLibrary",
		"targetPath": "app/src/main/jniLibs/arm64-v8a",
		"targetName": "main",
		"dependencies": {
			"bindbc-sdl": "~>1.4.7",
			"arsd-official:jni": "~>11.4.2"
		},
		"libs": ["android"],
		"subConfigurations": {
			"bindbc-sdl": "dynamicBC",
		},
		"versions": [ "SDL_200" ],
	}]
}
```
I then cross compile for android using LDC2 (after setting up the correct environment, [here](https://wiki.dlang.org/Build_D_for_Android)):

```sh
 dub --compiler=ldc2 --arch=aarch64-*-linux-android --config=android-64
```

**Comment by ichordev** [comment]
> ```
> ../../.dub/packages/bindbc-common/0.1.5/bindbc-common/source/bindbc/common/codegen.d(420,8):
> Error: static assert:  "Unknown runtime, not sure what mangling to use.
> Please check how your compiler mangles C++ struct constructors and
> add a case for it to `bindbc.common.codegen.mangleofCppDefaultCtor`."
> ```

This is triggered by code for making C++ bindings. Since D doesn't allow explicit default struct constructors but C++ does, I have to create the manglings manually with that function. Technically you're not using it, but I would prefer to just plug this missing case if possible.

It [sounds like](https://developer.android.com/ndk/guides/abis) Android requires C++ code to use the Itanium ABI, is that correct?

**Comment by DannyArends** [comment]
I think so, but I am kind of unfamiliar with the internals of how to bind function calls. I know for my setup it used to work fine on Android when using the 0.3.2 bindbc-loader.

**Comment by ichordev** [comment]
I have published v0.1.6, which should alleviate your issue.

**Comment by DannyArends** [comment]
I'll check it out and let you know if I run into any other issues
